core/shared/src/main/scala/cats/effect/IO.scala

```diff
@@ -862,17 +862,51 @@ private[effect] abstract class IOInstances extends IOLowPriorityInstances {
 }
 
 /**
- * @define shiftDesc For example we can introduce an asynchronous
+ * @define shiftDesc Note there are 2 overloads of the `IO.shift` function:
+ *
+ *         - One that takes a `ContextShift` that manages the
+ *           thread-pool used to trigger async boundaries.
+ *         - Another that takes a Scala `ExecutionContext` as the
+ *           thread-pool.
+ *
+ *         Please use the former by default and use the latter
+ *         only for fine-grained control over the thread pool in
+ *         use.
+ *
+ *         By default, Cats Effect can provide instance of
+ *         `ContextShift[IO]` that manages thread-pools, but
+ *         only if there’s an `ExecutionContext` in scope or
+ *         if `IOApp` is used:
+ *
+ *         {{{
+ *           import cats.effect.{IO, ContextShift}
+ *           import scala.concurrent.ExecutionContext.Implicits.global
+ *
+ *           val contextShift = IO.contextShift(global)
+ *         }}}
+ *
+ *         For example we can introduce an asynchronous
  *         boundary in the `flatMap` chain before a certain task:
  *         {{{
+ *           val task = IO(println("task"))
+ *
+ *           IO.shift(contextShift).flatMap(_ => task)
+ *         }}}
+ *
+ *         Note that the ContextShift value is taken implicitly
+ *         from the context so you can just do this:
+ *
+ *         {{{
  *           IO.shift.flatMap(_ => task)
  *         }}}
  *
  *         Or using Cats syntax:
  *         {{{
- *           import cats.syntax.all._
+ *           import cats.syntax.apply._
  *
  *           IO.shift *> task
+ *           // equivalent to
+ *           implicitly[ContextShift[IO]].shift *> task
```

Comment:
```suggestion
 *           ContextShift[IO].shift *> task
```

Since 1.1, we have an implicit summoner for `ContextShift` on the companion object, and I would consider its usage to be better practice than `implicitly`.

Reply:
Ah, yes I forgot about that. I'll send a PR that addresses its usage in the docs also.